### ISSUE TITLE
Now Playing banner vanishes after a brief display

### ISSUE THREAD

**Issue by lgo51** [fatg3erman/RompR#84]
Subject says it... see attached image...
RompR v1.32
![RompR issue1](https://user-images.githubusercontent.com/53191768/70394095-c0ed4300-19ae-11ea-809a-abb02c5d230b.jpg)

Fedora 25 
PHP 7.0.25 (cli) (built: Oct 28 2017 06:17:04) ( NTS )
Zend Engine v3.0.0, Copyright (c) 1998-2017 Zend Technologies
nginx version: nginx/1.12.1

**Comment by fatg3erman** [comment]
Going to need significantly more information to track that down. Please attach the output from the 'Debug Information' plugin. - Alhough PHP 7.0 is really really old and if it turns out to be that there's nothing I can do about it - but send the debug info anyway and I'll see if I can reproduce it.

**Comment by lgo51** [comment]
Sure... here 'tis...

* **BACKEND INFO**
* **Version** 1.32
* **mysql_database** [Redacted]
* **mysql_host** [Redacted]
* **mysql_password** [Redacted]
* **mysql_port** [Redacted]
* **mysql_user** [Redacted]
* **proxy_host**
* **proxy_password**
* **proxy_user**
* **spotify_token** [Redacted]
* **spotify_token_expires** [Redacted]
* **SERVER INFO**
* **PHP_SELF** /rompr/utils/debuginfo.php
* **SERVER_ADDR** 10.1.0.110
* **SERVER_NAME** antipodes.dev
* **SERVER_SOFTWARE** nginx/1.12.1
* **DOCUMENT_ROOT** /usr/share/nginx/html
* **HTTP_HOST** 10.1.0.110
* **HTTP_USER_AGENT** Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/78.0.3904.99 Safari/537.36 Vivaldi/2.9.1705.41
* **REMOTE_ADDR** 10.1.0.204
* **SERVER_PORT** 80
* **REQUEST_URI** /rompr/utils/debuginfo.php
* **COOKIES**
* **sortbydate** true
* **skin** desktop
* **currenthost** antipodes
* **player_backend** mpd
* **PHP INFO**
* **Version** 7.0.25
* **mbstring** 7.0.25
* **PDO** 7.0.25
* **pdo_mysql**
* **pdo_sqlite** 7.0.25
* **curl** 7.0.25
* **date** 7.0.25
* **fileinfo** 1.0.5
* **json** 1.4.0
* **SimpleXML** 7.0.25
* **GD** 7.0.25
* **GD Info** GD Version=2.2.5, FreeType Support=1, FreeType Linkage=with freetype, GIF Read Support=1, GIF Create Support=1, JPEG Support=1, PNG Support=1, WBMP Support=1, XPM Support=1, XBM Support=1, WebP Support=1, JIS-mapped Japanese Font Support=
* **ImageMagick** Not Installed
* **date.timezone** global_value=, local_value=, access=7
* **default_charset** global_value=UTF-8, local_value=UTF-8, access=7
* **default_socket_timeout** global_value=60, local_value=60, access=7
* **display_errors** global_value=, local_value=, access=7
* **error_log** global_value=/var/log/php-fpm/www-error.log, local_value=/var/log/php-fpm/www-error.log, access=4
* **error_reporting** global_value=22527, local_value=22527, access=7
* **file_uploads** global_value=1, local_value=1, access=4
* **log_errors** global_value=0, local_value=0, access=4
* **max_execution_time** global_value=1800, local_value=1800, access=7
* **mbstring.language** global_value=neutral, local_value=neutral, access=7
* **memory_limit** global_value=128M, local_value=128M, access=7
* **pdo_mysql.default_socket** [NOT SET]
* **session.use_cookies** global_value=1, local_value=1, access=7
* **upload_tmp_dir** global_value=, local_value=, access=4
* **PLAYER INFORMATION**
* **Connection Status** Connection Successful
* **tagtype** Artist, ArtistSort, Album, AlbumSort, AlbumArtist, AlbumArtistSort, Title, Track, Name, Genre, Date, Composer, Performer, Disc, MUSICBRAINZ_ARTISTID, MUSICBRAINZ_ALBUMID, MUSICBRAINZ_ALBUMARTISTID, MUSICBRAINZ_TRACKID, MUSICBRAINZ_RELEASETRACKID
* **Commands** add, addid, addtagid, channels, clear, clearerror, cleartagid, close, commands, config, consume, count, crossfade, currentsong, decoders, delete, deleteid, disableoutput, enableoutput, find, findadd, idle, kill, list, listall, listallinfo, listfiles, listmounts, listplaylist, listplaylistinfo, listplaylists, load, lsinfo, mixrampdb, mixrampdelay, mount, move, moveid, next, notcommands, outputs, password, pause, ping, play, playid, playlist, playlistadd, playlistclear, playlistdelete, playlistfind, playlistid, playlistinfo, playlistmove, playlistsearch, plchanges, plchangesposid, previous, prio, prioid, random, rangeid, readcomments, readmessages, rename, repeat, replay_gain_mode, replay_gain_status, rescan, rm, save, search, searchadd, searchaddpl, seek, seekcur, seekid, sendmessage, setvol, shuffle, single, stats, status, sticker, stop, subscribe, swap, swapid, tagtypes, toggleoutput, unmount, unsubscribe, update, urlhandlers, volume
* **URL Handlers** http://
* **URL Handlers** https://
* **URL Handlers** mms://
* **URL Handlers** mmsh://
* **URL Handlers** mmst://
* **URL Handlers** mmsu://
* **URL Handlers** gopher://
* **URL Handlers** rtp://
* **URL Handlers** rtsp://
* **URL Handlers** rtmp://
* **URL Handlers** rtmpt://
* **URL Handlers** rtmps://
* **URL Handlers** alsa://
* **plugin** mad, mpg123, oggflac, flac, dsdiff, dsf, faad, ffmpeg, pcm
* **suffix** mp3, mp2, mp3, ogg, oga, flac, dff, dsf, aac, 16sv, 3g2, 3gp, 4xm, 8svx, aa3, aac, ac3, adx, afc, aif, aifc, aiff, al, alaw, amr, anim, apc, ape, asf, atrac, au, aud, avi, avm2, avs, bap, bfi, c93, cak, cin, cmv, cpk, daud, dct, divx, dts, dv, dvd, dxa, eac3, film, flac, flc, fli, fll, flx, flv, g726, gsm, gxf, iss, m1v, m2v, m2t, m2ts, m4a, m4b, m4v, mad, mj2, mjpeg, mjpg, mka, mkv, mlp, mm, mmf, mov, mp+, mp1, mp2, mp3, mp4, mpc, mpeg, mpg, mpga, mpp, mpu, mve, mvi, mxf, nc, nsv, nut, nuv, oga, ogm, ogv, ogx, oma, ogg, omg, opus, psp, pva, qcp, qt, r3d, ra, ram, rl2, rm, rmvb, roq, rpl, rvc, shn, smk, snd, sol, son, spx, str, swf, tak, tgi, tgq, tgv, thp, ts, tsp, tta, xa, xvid, uv, uv2, vb, vid, vob, voc, vp6, vmd, wav, webm, wma, wmv, wsaud, wsvga, wv, wve
* **mime_type** audio/mpeg, application/ogg, application/x-ogg, audio/ogg, audio/x-flac+ogg, audio/x-ogg, application/flac, application/x-flac, audio/flac, audio/x-flac, application/x-dff, audio/x-dff, audio/x-dsd, application/x-dsf, audio/x-dsf, audio/x-dsd, audio/aac, audio/aacp, application/flv, application/m4a, application/mp4, application/octet-stream, application/ogg, application/x-ms-wmz, application/x-ms-wmd, application/x-ogg, application/x-shockwave-flash, application/x-shorten, audio/8svx, audio/16sv, audio/aac, audio/aacp, audio/ac3, audio/aiffaudio/amr, audio/basic, audio/flac, audio/m4a, audio/mp4, audio/mpeg, audio/musepack, audio/ogg, audio/opus, audio/qcelp, audio/vorbis, audio/vorbis+ogg, audio/x-8svx, audio/x-16sv, audio/x-aac, audio/x-ac3, audio/x-adx, audio/x-aiffaudio/x-alaw, audio/x-au, audio/x-dca, audio/x-eac3, audio/x-flac, audio/x-gsm, audio/x-mace, audio/x-matroska, audio/x-monkeys-audio, audio/x-mpeg, audio/x-ms-wma, audio/x-ms-wax, audio/x-musepack, audio/x-ogg, audio/x-vorbis, audio/x-vorbis+ogg, audio/x-pn-realaudio, audio/x-pn-multirate-realaudio, audio/x-speex, audio/x-ttaaudio/x-voc, audio/x-wav, audio/x-wma, audio/x-wv, video/anim, video/quicktime, video/msvideo, video/ogg, video/theora, video/webm, video/x-dv, video/x-flv, video/x-matroska, video/x-mjpeg, video/x-mpeg, video/x-ms-asf, video/x-msvideo, video/x-ms-wmv, video/x-ms-wvx, video/x-ms-wm, video/x-ms-wmx, video/x-nut, video/x-pva, video/x-theora, video/x-vid, video/x-wmv, video/x-xvid, audio/x-mpd-ffmpeg, audio/L16, audio/x-mpd-float, audio/x-mpd-cdda-pcm, audio/x-mpd-cdda-pcm-reverse
* **CONFIG**
* **rgbs** r=100, g=50, b=1, a=1
* **music_directory_albumart** /storage/music/
* **sortbycomposer** true
* **composergenre** true
* **composergenrename** 0=Classical
* **preferlocalfiles** true
* **mopidy_collection_folders** 0=Spotify Playlists, 1=Local media, 2=SoundCloud/Liked
* **lastfm_country_code** US
* **country_userset** true
* **debug_enabled** 0
* **custom_logfile**
* **cleanalbumimages** true
* **do_not_show_prefs** false
* **load_plugins_at_loadtime** false
* **beets_server_location**
* **multihosts** antipodes=[host=localhost, port=6600, password=, socket=, mopidy_slave=false, radioparams=[radiomode=, radioparam=, radiomaster=, radioconsume=0]], DSx=[host=10.1.0.108, port=6600, password=, socket=, mopidy_slave=false, radioparams=[radiomode=, radioparam=, radiomaster=, radioconsume=0]], default=[host=localhost, port=6600, password=, socket=, mopidy_slave=false, radioparams=[radiomode=, radioparam=, radiomaster=, radioconsume=0]]
* **dev_mode** false
* **live_mode** false
* **collection_load_timeout** 3600000
* **smartradio_chunksize** 6
* **linkchecker_nextrun** 1576141943182
* **linkchecker_isrunning** false
* **linkchecker_frequency** 604800000
* **linkchecker_polltime** 5000
* **audiobook_directory** /storage/ebook/
* **collection_player** mpd
* **snapcast_server**
* **snapcast_port** 1705
* **displaycomposer** true
* **artistsatstart** 0=Various Artists, 1=Soundtracks
* **nosortprefixes** 0=The
* **sortcollectionby** albumbyartist
* **showartistbanners** true
* **google_api_key**
* **google_search_engine_id**
* **sync_lastfm_playcounts** false
* **sync_lastfm_at_start** false
* **last_lastfm_synctime** 1575531257
* **lfm_importer_start_offset** 0
* **lfm_importer_last_import** 0
* **sortbydate** true
* **notvabydate** false
* **currenthost** antipodes
* **player_backend** mpd
* **collectionrange** 0
* **lastfm_user**
* **lastfm_session_key**
* **autotagname**
* **tradsearch** false
* **lastfm_scrobbling** false
* **lastfm_autocorrect** true
* **sourceshidden** false
* **playlisthidden** false
* **infosource** discogs
* **playlistcontrolsvisible** false
* **sourceswidthpercent** 23.136002234959545
* **playlistwidthpercent** 23.36611427868913
* **downloadart** true
* **clickmode** double
* **chooser** albumlist
* **hide_albumlist** false
* **hide_filelist** false
* **hide_radiolist** false
* **hide_podcastslist** false
* **hide_playlistslist** false
* **hide_audiobooklist** false
* **hide_searcher** false
* **hidebrowser** false
* **shownupdatewindow** 1.32
* **scrolltocurrent** true
* **alarm_ramptime** 30
* **alarm_snoozetime** 8
* **lastfmlang** default
* **user_lang** en
* **synctags** false
* **synclove** false
* **synclovevalue** 5
* **theme** Curves.css
* **icontheme** Colourful
* **coversize** 40-Large.css
* **fontsize** 05-Huge.css
* **fontfamily** Nunito.css
* **collectioncontrolsvisible** false
* **displayresultsas** tree
* **crossfade_duration** 2
* **newradiocountry** countries/US
* **search_limit_limitsearch** false
* **scrobblepercent** 50
* **updateeverytime** false
* **fullbiobydefault** true
* **mopidy_search_domains** 0=local, 1=spotify
* **mopidy_radio_domains** 0=local, 1=spotify
* **outputsvisible** false
* **wheelscrollspeed** 150
* **searchcollectiononly** false
* **displayremainingtime** true
* **cdplayermode** false
* **auto_discovembobulate** false
* **ratman_sortby** Rating
* **ratman_showletters** false
* **sleeptime** 30
* **sleepon** false
* **advanced_search_open** false
* **sortwishlistby** artist
* **player_in_titlebar** true
* **communityradiocountry** united states
* **communityradiolanguage**
* **communityradiotag**
* **communityradiolistby** country
* **communityradioorderby** name
* **browser_id** 1575531257628
* **playlistswipe** true
* **podcastcontrolsvisible** true
* **default_podcast_display_mode** 0
* **default_podcast_refresh_mode** 3
* **default_podcast_sort_mode** 0
* **podcast_mark_new_as_unlistened** false
* **use_albumart_in_playlist** true
* **podcast_sort_levels** 4
* **podcast_sort_0** Title
* **podcast_sort_1** Artist
* **podcast_sort_2** Category
* **podcast_sort_3** new
* **bgimgparms** dummy=baby
* **alarms**
* **collection_type** sqlite
* **test_width** 1578
* **test_height** 971
* **skin** desktop
* **maxrgbs** r=255, g=75, b=1, a=1
* **bgdrgbs** r=0, g=0, b=0, a=0
* **radiomode**
* **radioparam**
* **radiomaster**
* **radioconsume** 0

**Comment by fatg3erman** [comment]
I cannot reproduce this at all. More info would help, like does it always do it? Is it only with certain tracks? Anything that might be helpful.

**Comment by lgo51** [comment]
Hi Mark,

For us, happens with: Edge, Vivaldi, Chrome, Firefox, Palemoon running on Win10 or LinuxMint19 or Android... interestingly, did not occur with Safari on iPad Air2 [latest OS].

We observe a track load, title banner shows up, then vanishes for remainder of track duration.

Theme setting did not seem to affect behavior.

Happens for all tracks.

Initial site loading, to arrive at operational screen, can take up to 5 minutes... and response to buttons is [uselessly] slow.

CPU/RAM utilization on web [nginx] server is trivially low, save during initial screen setup when 1 of 4 cores might be at 80%

Hope this helps.

Cheers,
Larry

**Comment by fatg3erman** [comment]
Thanks Larry, there are some clues there.

Some further questions:

Is the screen resolution of your device very high? Try resizing your browser window to a much smaller size and see if that makes a difference. Is this just on the Desktop skin, or do all the skins do it - for example in a browser that is exhibiting the problem try also the Skypotato skin and the Phone skin. Also try a different font.

The slow loading is probably not related but is also concerning. 80% CPU usage even on one core is troublingly high. Are you subscribed to a lot of podcasts? What device is your webserver running on?

**Comment by lgo51** [comment]
> On December 28, 2019 at 3:15 PM Mark Greenwood <notifications@github.com> wrote:
> 
> 
>     Thanks Larry, there are some clues there.
> 
>     Some further questions:
> 
>     Is the screen resolution of your device very high?
> 
some 1920x1080, some 1366x768 some 1280x720

> 
>     Try resizing your browser window to a much smaller size and see if that makes a difference.
> 
did not try

> Is this just on the Desktop skin, or do all the skins do it - for example in a browser that is exhibiting the problem try also the Skypotato skin and the Phone skin.
> 
all skins [mis]behave the same

> 
>     Also try a different font.
> 
did not try

> 
>     The slow loading is probably not related but is also concerning. 80% CPU usage even on one core is troublingly high.
> 
agreed

> 
>     Are you subscribed to a lot of podcasts?
> 
zero

> 
>     What device is your webserver running on?
> 
Gen2 i5 laptop w/ 8GB RAM
Gen4/5 Pentium w/ 8GB RAM
Fedora 25 [text mode] / nginx

very [tediously] slow GUI seems to be unrelated to CPU load (can be seen for sure, but plenty of headroom) and RAM is far from maxed

Clue: track-time counters often jump by 2 - 9 seconds as screen is that far behind most of the time

**Comment by fatg3erman** [comment]
OK, so it definitely looks like somethng is causing your browser to grind to a halt, that would explain everything you're seeing.

First definitely try sorting by Artist and not Albums (by Artist). The former is much lighter on browser resources, especially if you have the option to automatically download album art enabled.

Secondly if that doesn't explain it I'm going to need to see a debug log (https://fatg3erman.github.io/RompR/Troubleshooting#debug-logging) at level 9.

**Comment by fatg3erman** [comment]
OK I've managed to track this one down.

Unfortunately it's your browser struggling to load and display all the album images when you use 'Albums (by Artist)' mode. This has always been an issue with browsers and most of them seem to be getting much worse very quickly - though Safari on iOS is still really good at it, oddly.

For now using 'Artist' mode will solve it. In the next release I've made some changes to 'lazy-load' the images only as-and-when they're needed rather than all at once. It requires a relatively modern browser but it's the only way.
